client/src/main/java/org/apache/rocketmq/client/ClientConfig.java

```diff
@@ -241,6 +247,15 @@ public void setLanguage(LanguageCode language) {
     }
 
     public String getNamespace() {
+        if (StringUtils.isNotEmpty(namespace)) {
+            return namespace;
+        }
+
+        if (StringUtils.isNotEmpty(this.namesrvAddr)) {
```

Comment:
Here,do you mean that if namespace is empty,then return the value by parseInstanceIdFromEndpoint() method?

Reply:
yep，we can get the namespace by parse Access point UR